hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/handshakeserver/HandshakeServer.java

```diff
@@ -44,163 +43,210 @@
+						ret_code = MessageUtil.HNET_RTRN_INVALIDFORMATERROR;
+						hnSecureResponse = ErrorBuilder
+								.buildDefaultErrorMessage(MessageUtil.HL7Error_Msg_ErrorDTHeaderToHNClient);
+					}
+
+				}
+
+				// Look for the start of data
+				if (headerIn.contains(DATA_INDICATOR)) {
+					int messageLength = Integer
+							.parseInt(headerIn.substring(DATA_INDICATOR_LENGTH, LENGTH_INDICATOR_LENGTH));
+
+					numSocketReadTries = 0;
+
+					while (socketInput.available() < messageLength) {
+						if (numSocketReadTries < MAX_SOCKET_READ_TRIES) {
+							numSocketReadTries++;
+							// Give other packets a chance to arrive and
+							// avoid trying to monopolize CPU time..
+							Thread.sleep(SOCKET_READ_SLEEP_TIME);
+						} else {
+							// Give up eventually.
+							ret_code = MessageUtil.HNET_RTRN_INVALIDFORMATERROR;
+							logger.debug("{} - Error receiving HL7 message from Listener: {}", methodName, ret_code);
+
+						}
+					}
+
+					logger.debug("{} - Received from originator {} byte DT Data Block: {}", methodName, messageLength,
+							ret_code);
+					// read dtsegment data
+					byte[] dtMessage = new byte[messageLength];
+					socketInput.read(dtMessage, 0, messageLength);
+					extractData(dtMessage);
+					String HL7IN = new String(dtMessage, StandardCharsets.UTF_8);
+					int indexOfMSG = HL7IN.indexOf(HEADER_INDICATOR);
+
+					if (indexOfMSG != -1) {
+						// Read whatever is after the MSH segment.
+						String aResponse = HL7IN.substring(indexOfMSG) + "\r";
+						HL7IN = aResponse;
+						logger.debug("{} - HL7 message recived from POS: {}", methodName, aResponse);
+					}
+
+					// Send received message to server
+					logger.debug("{} - HL7 message received from POS: {}", methodName, HL7IN);
+					try {
+						logger.info("{} - Attempting to send the txn to remote server: {}", methodName, ret_code);
+						hnSecureResponse = (String) producer.requestBody(HL7IN);
+					} catch (Exception e) {
+						ret_code = MessageUtil.HNET_RTRN_REMOTETIMEOUT;
+						logger.debug("{} - Failed to send request to remote server:{}", methodName, ret_code);
+						hnSecureResponse = ErrorBuilder
+								.buildHTTPErrorMessage(MessageUtil.HL7Error_Msg_ServerUnavailable, null);
+					}
+
+				} else {
+					logger.debug("{} - Error recieving DT segment: {}", methodName,
+							MessageUtil.HL7Error_Msg_ErrorDTHeaderToHNClient);
+					hnSecureResponse = ErrorBuilder
+							.buildDefaultErrorMessage(MessageUtil.HL7Error_Msg_ErrorDTHeaderToHNClient);
+				}
+
+				// Write Response back to POS
+				if (StringUtil.isNullOrEmpty(hnSecureResponse)) {
+					hnSecureResponse = ErrorBuilder
+							.buildDefaultErrorMessage(MessageUtil.HL7Error_Msg_ServerUnavailable);
 				}
+				sendResponse(socketOutput, hnSecureResponse);
+
+				// sent Response to POS
+				logger.info("{} - HL7 transaction is done: {} {}", methodName, ret_code, System.lineSeparator());
+				decodeSeed = 0;
```

Comment:
Why reseting? did the value changed anywhere?

Reply:
Yes, this need to reset for the request.